libtiledbsoma/src/soma/soma_sparse_ndarray.h

```diff
@@ -157,6 +157,14 @@ class SOMASparseNDArray : public SOMAArray {
      * @return std::unique_ptr<ArrowSchema>
      */
     std::unique_ptr<ArrowSchema> schema() const;
+
+    /**
+     * @brief Get the soma_data's dtype in the form of an Arrow
+     * format string.
+     *
+     * @return std::string_view Arrow format string.
+     */
+    std::string_view soma_data_type();
```

Comment:
From glancing at it, both functions appear to be identical so should they be promoted 'up' to SOMAArray?

Reply:
The problem with that is `SOMADataFrame` does not use `soma_data`

Reply:
Spot on, thank you. I thought of that (but didn't open the file).  Make sense as is then.